experimental/dds/tree2/src/core/rebase/types.ts

```diff
@@ -24,6 +24,33 @@ export const SessionIdSchema = brandedStringType<SessionId>();
 export type RevisionTag = StableId;
 export const RevisionTagSchema = brandedStringType<StableId>();
 
+/**
+ * An ID which is unique within a revision of a changeset.
+ * A changeset which is a composition of multiple revisions may contain duplicate `ChangesetLocalId`s,
+ * but they are unique when qualified by the revision of the change they are used in.
+ * @alpha
+ */
+export type ChangesetLocalId = Brand<number, "ChangesetLocalId">;
```

Comment:
Moving this would have been better done as a separate PR to reduce the noise and avoid the merge conflicts. Up to you if you want to keep it bundled or not.